VideoAPI/Video.API/Controllers/ConsultationController.cs

```diff
@@ -283,6 +283,38 @@ public async Task<IActionResult> StartConsultationRequestAsync(StartConsultation
             }
         }
 
+        /// <summary>
+        /// Leave a consultation.
+        /// </summary>
+        /// <param name="request"></param>
+        /// <returns></returns>
+        [HttpPost("end")]
+        [SwaggerOperation(OperationId = "LeaveConsultationAsync")]
+        [ProducesResponseType((int)HttpStatusCode.OK)]
+        [ProducesResponseType((int)HttpStatusCode.NotFound)]
+        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+        public async Task<IActionResult> LeaveConsultationAsync(LeaveConsultationRequest request)
+        {
+            var getConferenceByIdQuery = new GetConferenceByIdQuery(request.ConferenceId);
+            var conference = await _queryHandler.Handle<GetConferenceByIdQuery, Conference>(getConferenceByIdQuery);
+            if (conference == null)
+            {
+                _logger.LogWarning("Unable to find conference");
+                return NotFound();
+            }
+            var participant = conference.GetParticipants().SingleOrDefault(x => x.Id == request.ParticipantId);
+            if (participant == null)
+            {
+                _logger.LogWarning("Unable to find participant request by id");
+                return NotFound();
+            }
+
+            var currentRoom = participant.CurrentVirtualRoom.Label;
+            await _consultationService.LeaveConsultationAsync(request, currentRoom,
```

Comment:
This method could return a KinlyApiException which needs to return a bad request similar to `StartConsultationRequestAsync`. See line 277

Reply:
updated to catch kinly api error